tests/integration/test_http_get_all.py

```diff
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import utils
+
+
+def test_http_get_all(test_env, default_request, tmpdir):
+    """
+    Check HTTP GET_ALL works as expected.
+
+    Process:
+    * Create at least two requests and wait for them to complete
+    * Send request to check all requests in Cachito at the /api/v1/requests API endpoint
+    * Send requests to check all requests with filtering parameters
+    Checks:
+    * Check that response code is 200
+    * Check that filtering parameters are working properly (page, per_page, and state)
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    requests_amount = test_env["http_get_all"]["requests_amount"]
+    submitted_requests = []
+    for _ in range(requests_amount):
+        initial_response = client.create_new_request(
+            payload={
+                "repo": test_env["package"]["repo"],
+                "ref": test_env["package"]["ref"],
+                "pkg_managers": test_env["package"]["pkg_managers"],
+            },
+        )
+        completed_response = client.wait_for_complete_request(initial_response)
+        response_data = completed_response.data
+        submitted_requests.append(response_data.get("id"))
+
+    assert_completed_requests(client, requests_amount)
+    assert_no_requests_in_progress_state(client, requests_amount, submitted_requests)
+
+
+def test_get_all_verbose(test_env, default_request, tmpdir):
+    """
+    Check HTTP GET_ALL with verbose works as expected.
+
+    Process:
+    * Create a request and wait for it to complete
+    * Send request get_all to the Cachito API with verbose
+    Checks:
+    * Check that response code is 200
+    * Check that verbose is working properly
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    assert_verbose_correct(client, test_env)
+
+
+def assert_completed_requests(client: utils.Client, requests_amount):
+    """Check get_all request returns expected completed values."""
+    request_json = {"per_page": requests_amount, "page": 1, "state": "complete"}
+    response = client.fetch_all_requests(request_json)
+    assert response.status == 200
+
+    assert all(item["state"] == "complete" for item in response.data.get("items"))
+    assert 0 <= len(response.data.get("items")) <= requests_amount
```

Comment:
Since it's a separate function, I get the appeal of trying to make it reusable for all situations but I'd really like `test_http_get_all` to verify that it is exactly `requests_amount`.
```suggestion
    assert len(response.data.get("items")) == requests_amount
```

Reply:
Thank you. Done.